src/Scan/BasePolling.spec.ts

```diff
@@ -0,0 +1,221 @@
+import 'reflect-metadata';
+import { Breakpoint } from './Breakpoint';
+import { Logger, logger } from '../Utils';
+import { BasePolling } from './BasePolling';
+import { Scans, ScanState, ScanStatus } from './Scans';
+import {
+  instance,
+  mock,
+  objectContaining,
+  reset,
+  spy,
+  verify,
+  when
+} from 'ts-mockito';
+
+describe('BasePolling', () => {
+  const breakpointMock = mock<Breakpoint>();
+  const scanManagerMock = mock<Scans>();
+
+  let loggerSpy!: Logger;
+
+  beforeEach(() => {
+    loggerSpy = spy(logger);
+  });
+
+  afterEach(() => {
+    reset<Breakpoint | Scans | Logger>(
+      breakpointMock,
+      scanManagerMock,
+      loggerSpy
+    );
+  });
+
+  describe('constructor', () => {
+    it('should warn if timeout is not specified', () => {
+      // arrange
+      const options = {
+        scanId: 'scanId'
+      };
+
+      // act
+      new BasePolling(
+        options,
+        instance(scanManagerMock),
+        instance(breakpointMock)
+      );
+
+      // assert
+      verify(
+        loggerSpy.warn(
+          `Warning: It looks like you've been running polling without "timeout" option.`
+        )
+      ).once();
+      verify(
+        loggerSpy.warn(
+          `The recommended way to install polling with a minimal timeout: 10-20min.`
+        )
+      ).once();
+    });
+
+    it('should warn if interval is less than 10s', () => {
+      // arrange
+      const options = {
+        scanId: 'scanId',
+        interval: 5000
+      };
+
+      // act
+      new BasePolling(
+        options,
+        instance(scanManagerMock),
+        instance(breakpointMock)
+      );
+
+      // assert
+      verify(loggerSpy.warn(`Warning: polling interval is too small.`)).once();
+      verify(
+        loggerSpy.warn(`The recommended way to set polling interval to 10s.`)
+      ).once();
+    });
+  });
+
+  describe('start', () => {
+    it('should start polling and stop on breakpoint exception', async () => {
+      // arrange
+      const scanId = 'scanId';
```

Comment:
Please ensure that the test data appears realistic (i.e. utilize the short-uuid).